### PR TITLE
fix out-of-date banner style

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/menubar/out-of-date.js [hunk 1]
```diff
@@ -48,12 +57,10 @@ const styles = Styles.styleSheetCreate({
     color: Styles.globalColors.white,
   },
   boxNonCritical: {
-    backgroundColor: Styles.globalColors.lightGrey,
+    backgroundColor: Styles.globalColors.yellow,
     padding: Styles.globalMargins.tiny,
```
Comment (adamjspooner) [inline shared/menubar/out-of-date.js]: Do we want to increase the top and bottom padding? Do we need to add a border radius to the top left and right corners?

File: shared/menubar/out-of-date.js [hunk 2]
```diff
@@ -21,15 +21,24 @@ const OutOfDate = ({outOfDate, updateNow}: Props) =>
       centerChildren={true}
       direction="vertical"
     >
-      <Kb.Text type="BodySemibold" style={outOfDate.critical ? styles.textCritical : styles.textNonCritical}>
+      <Kb.Text
+        backgroundMode="Information"
+        type="BodySemibold"
+        style={outOfDate.critical ? styles.textCritical : undefined}
```
[the remark this answers is not in the document]
Comment (songgao) [reply shared/menubar/out-of-date.js]: I tried that initially but flow was complaining that style isn't compatible with a boolean.

File: shared/menubar/out-of-date.js [hunk 3]
```diff
@@ -48,12 +57,10 @@ const styles = Styles.styleSheetCreate({
...
   },
   boxNonCritical: {
-    backgroundColor: Styles.globalColors.lightGrey,
+    backgroundColor: Styles.globalColors.yellow,
     padding: Styles.globalMargins.tiny,
   },
-  textNonCritical: {
-    color: Styles.globalColors.black,
-  },
+  textNonCritical: {},
```
[the remark this answers is not in the document]
Comment (songgao) [reply shared/menubar/out-of-date.js]: Nope; forgot to delete it

File: shared/menubar/out-of-date.js (hunk 1 above)
Comment (songgao) [reply shared/menubar/out-of-date.js]: This is how the margin *should* look, without the occasional cutoff bug: 
screen shot 2018-10-08 at 10 35 26 am

Do you mean we should make it bigger than this?

File: shared/menubar/out-of-date.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamjspooner) [reply shared/menubar/out-of-date.js]: Really? I just tested locally and it worked okay (though, on a different component).

[the remark this answers is not in the document]
Comment (songgao) [reply shared/menubar/out-of-date.js]: Hrm; let me try again; maybe I mis-read the flow error.

File: shared/menubar/out-of-date.js (hunk 1 above)
Comment (adamjspooner) [reply shared/menubar/out-of-date.js]: Nope! That looks good to me. What causes the cutoff bug?

File: shared/menubar/out-of-date.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cecileboucheron) [reply shared/menubar/out-of-date.js]: @songgao @adamjspooner Banners are using BodySmallSemibold now

[the remark this answers is not in the document]
Comment (songgao) [reply shared/menubar/out-of-date.js]: OK will update!

File: shared/menubar/out-of-date.js (hunk 1 above)
Comment (songgao) [reply shared/menubar/out-of-date.js]: Haven't looked into that yet. Maybe it's part of the layout thing that John mentioned in #frontend , i.e. that the banner should be part of the blue header in layout

File: shared/menubar/out-of-date.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (songgao) [reply shared/menubar/out-of-date.js]: @adamjspooner Yeah; still not working:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ menubar/out-of-date.js:41:18
Cannot create Kb.Text element because in property style:
 • Either inexact boolean [1] is incompatible with exact _StylesCrossPlatform [2].
 • Or boolean [1] is incompatible with read-only array type [3].
     menubar/out-of-date.js
     34│         style={outOfDate.critical && styles.textCritical}
     35│       >
     36│         Please{' '}
     37│         <Kb.Text
     38│           backgroundMode="Information"
     39│           type="BodySemibold"
     40│           underline={!!updateNow}
     41│           style={outOfDate.critical && styles.textCritical}
     42│           onClick={updateNow}
     43│         >
     44│           update now
     45│         </Kb.Text>
     46│         .
     47│       </Kb.Text>
     48│     </Kb.Box2>
     constants/types/config.js
 [1] 10│   critical: boolean,
     styles/css.js.flow
 [2] 96│   | _StylesCrossPlatform
 [3] 97│   | $ReadOnlyArray<_StylesCrossPlatform | null | false | void>
```

[the remark this answers is not in the document]
Comment (adamjspooner) [reply shared/menubar/out-of-date.js]: Weird. I don't fully understand what an `inexact boolean` is. I thought that related to objects: https://devhints.io/flow#objects

[the remark this answers is not in the document]
Comment (songgao) [reply shared/menubar/out-of-date.js]: I think it's just that the only way `style` can take a boolean is when it's part of an array. i.e. it's OK to have `style={[false]}`, but not `style={false}`.

[the remark this answers is not in the document]
Comment (adamjspooner) [reply shared/menubar/out-of-date.js]: I use this pattern (`style={props.someBoolean && styles.appliedStylesIfTrue`) all over the place, though. @chrisnojima, can you explain what's going on here?

**Comment by adamjspooner** [comment]
Don't forget to update your storyshots.

File: shared/menubar/out-of-date.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/menubar/out-of-date.js]: That works only in the contexts of collapseStyles (we explicitly allow Boolean and filter it out)

[the remark this answers is not in the document]
Comment (adamjspooner) [reply shared/menubar/out-of-date.js]: Thanks! I'm sorry for the confusion, @songgao!

[the remark this answers is not in the document]
Comment (adamjspooner) [reply shared/menubar/out-of-date.js]: 🤔 So, which is the preferred method?

```
style={Styles.collapseStyles([someBoolean && someStyles])}
```
or:
```
style={someBoolean ? someStyles : undefined}
```

[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/menubar/out-of-date.js]: Either is fine. CollapseStyles is to mix multiple styles so it's a little weird to use it in this case

[the remark this answers is not in the document]
Comment (songgao) [reply shared/menubar/out-of-date.js]: NP! Thanks for the discussion and clarification.
